tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -320,6 +373,41 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 						rawValue: runs,
 						formattedValue: prettyNumber(runs, 0),
 					};
+
+					if (baselineMemoryUsage >= 0 && allowedDeviationBytes >= 0) {
```

Comment:
Looks like there are some duplications. More readable way is like:
```suggestion
					if (avgHeapUsed > upperBound) {
        reportMemoryIssue(
            `Memory regression detected for test '${testTitle}': Used '${avgHeapUsed.toPrecision(
                6,
            )}' bytes, baseline '${baselineMemoryUsage}', tolerance '${allowedDeviationBytes}' bytes.\n`,
        );
    } else if (avgHeapUsed < lowerBound) {
        reportMemoryIssue(
            `Possible memory improvement detected for test '${testTitle}': Used '${avgHeapUsed.toPrecision(
                6,
            )}' bytes, baseline '${baselineMemoryUsage}', tolerance '${allowedDeviationBytes}' bytes. Consider updating the baseline.\n`,
        );
    }
```

Reply:
```suggestion
function reportMemoryIssue(message: string): void {
    if (ENABLE_MEM_REGRESSION) {
        throw new Error(message);
    } else {
        process.stdout.write(chalk.yellow(message));
    }
}
```

Reply:
Great idea, thanks!